### PR TITLE
Changed price bucket calculation logic to start increment at bucketMin

### PR DESCRIPTION
**Pull request by Pubmatic-Dhruv-Sonone**
Fix for https://github.com/prebid/prebid-server/issues/2469

### CODE REVIEW

File: exchange/price_granularity.go [hunk 1]
```diff
@@ -13,14 +13,18 @@ func GetPriceBucket(cpm float64, config openrtb_ext.PriceGranularity) string {
...
 	for i := 0; i < len(config.Ranges); i++ {
 		if config.Ranges[i].Max > bucketMax {
 			bucketMax = config.Ranges[i].Max
 		}
 		// find what range cpm is in
 		if cpm >= config.Ranges[i].Min && cpm <= config.Ranges[i].Max {
 			increment = config.Ranges[i].Increment
+			bucketMin = bucketFloor
+		} else {
+			bucketFloor = config.Ranges[i].Max
```
Comment (mansinahar) [inline exchange/price_granularity.go:27]: You don't need to keep track of the prev max and assign that to `bucketMin`. This is already done [here](https://github.com/prebid/prebid-server/blob/master/openrtb_ext/request.go#L241). You can just simply keep track of the smallest `bucketMin`, similar to how `bucketMax` is tracked and pass that value to the `getCpmTarget` function. With this change you won't need `bucketFloor` at all.

Something like:

```
	bucketMax := 0.0
	bucketMin := math.MaxFloat64
	increment := 0.0
	precision := config.Precision

	for i := 0; i < len(config.Ranges); i++ {
		if config.Ranges[i].Min < bucketMin {
			bucketMin = config.Ranges[i].Min
		}
		if config.Ranges[i].Max > bucketMax {
			bucketMax = config.Ranges[i].Max
		}
```

Comment (Pubmatic-Dhruv-Sonone) [reply exchange/price_granularity.go:27]: @mansinahar 
We want to keep track for the bucketMin  for the range from which increment is selected, not the smallest bucketMin. smallest bucketMin will always be zero as done [here](https://github.com/prebid/prebid-server/blob/f29682a87293010a346aeed4a5870b4fe4c37eee/openrtb_ext/request.go#L232).

Comment (mansinahar) [reply exchange/price_granularity.go:27]: @Pubmatic-Dhruv-Sonone Ah! You're right. That was my mistake. So you only need bucketMin = config.Ranges[i].Min on L25 and we can remove bucketFloor altogether as the validation code already makes sure that the previous max is set as min for the next range. Can you please make that change? Thanks!

Comment (Pubmatic-Dhruv-Sonone) [reply exchange/price_granularity.go]: Done. Made the change to set ` bucketMin = config.Ranges[i].Min`

**Review by hhhjort — APPROVED** [review]
LGTM

**Review by mansinahar — APPROVED** [review]

**Pull request merged**